### PR TITLE
Tweak cross-compilation of some netbsd stuff

### PR DESCRIPTION
**Pull request by kirelagin**
###### Motivation for this change

_I have no idea what I am doing._

I am dead set on getting systemd compiled with musl (almost there), and these things stood on my way, so I had to do _something_.

###### Things done

- [ ] Tested using sandboxing ([nix.useSandbox](http://nixos.org/nixos/manual/options.html#opt-nix.useSandbox) on NixOS, or option `sandbox` in [`nix.conf`](http://nixos.org/nix/manual/#sec-conf-file) on non-NixOS linux)
- Built on platform(s)
   - [x] NixOS
   - [x] macOS
   - [ ] other Linux distributions
- [ ] Tested via one or more NixOS test(s) if existing and applicable for the change (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
- [ ] Tested compilation of all pkgs that depend on this change using `nix-shell -p nix-review --run "nix-review wip"`
- [x] Tested execution of all binary files (usually in `./result/bin/`)
- [ ] Determined the impact on package closure size (by running `nix path-info -S` before and after)
- [x] Ensured that relevant documentation is up to date
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/.github/CONTRIBUTING.md).

###### Notify maintainers

cc @matthewbauer or @Ericson2314 or I don’t even know

### CODE REVIEW

**Comment by kirelagin** [comment]
The testing strategy is to do:

```
nix-build '<nixpkgs>' -A netbsd.getent --arg crossSystem '(import <nixpkgs/lib>).systems.examples.musl64'
```

and see that it did not work before and now it works.

**Comment by Ericson2314** [comment]
I hope there is another way to do this! haha. Pinging @matthewbauer who made all these *for* musl, as it turns out, and also taking a peak myself.

File: pkgs/os-specific/bsd/netbsd/default.nix [hunk 1]
```diff
@@ -127,6 +127,10 @@ let
     nativeBuildInputs = [ makeMinimal ];
     buildInputs = [ zlib ];
 
+    # for some reason the build system re-runs configure with HOST_CC
```
Comment (Ericson2314) [inline pkgs/os-specific/bsd/netbsd/default.nix]:
```suggestion
    # The build for some reason system re-runs configure with `HOST_CC` to make
    # some `defs.mk` file which is installed.
```

File: pkgs/os-specific/bsd/netbsd/default.nix [hunk 2]
```diff
@@ -127,6 +127,10 @@ let
...
+    depsBuildBuild = [ buildPackages.stdenv.cc ] ++ buildInputs;
+    HOST_CC = "${buildPackages.stdenv.cc}/bin/cc";
```
Comment (Ericson2314) [inline pkgs/os-specific/bsd/netbsd/default.nix]: Shouldn't need full path; it doesn't hurt but this matches the above.
```suggestion
    HOST_CC = "${buildPackages.stdenv.cc.targetPrefix}cc";
```

File: pkgs/os-specific/bsd/netbsd/default.nix [hunk 3]
```diff
@@ -211,16 +215,17 @@ let
     version = "8.0";
     nativeBuildInputs = [ ];
     propagatedBuildInputs = [ compat ];
+    buildInputs = [ stdenv.cc ];
```
Comment (Ericson2314) [inline pkgs/os-specific/bsd/netbsd/default.nix]: I don't think this is needed?

**Review by Ericson2314 — COMMENTED** [review]
OK I read the netbsd source and now I see the madness and understand your response :) Looks basically all good just a few quibbles.

**Review by Ericson2314 — APPROVED** [review]
Looks good!

**Comment by kirelagin** [comment]
(I came back to say that I have _just_ tested it on darwin and it still works.)

File: pkgs/os-specific/bsd/netbsd/default.nix (hunk 3 above)
Comment (kirelagin) [reply pkgs/os-specific/bsd/netbsd/default.nix]: That’s like _THE MOST ESSENTIAL PART_  of all it. It makes sure the cc-wrapper’s setup-hook runs and makes buildBuild deps (in this case zlib) available for the buildBuild compiler.

Comment (kirelagin) [reply pkgs/os-specific/bsd/netbsd/default.nix]: Oh, sorry, I was thinking about a different line. Let me think about this one.

**Comment by Ericson2314** [comment]
oh, whew!

**Pull request merged**